lib/jobs/ipmi-job.js

```diff
@@ -184,20 +184,117 @@ function ipmiJobFactory(
+        var self = this;
+        var stringifiedEntries = "";
+        var lastUpdatedSelDate;
+        var currentSelDate;
+        var lastReportedSelEntryID;
+        var totalEntries;
+        var lastUpdatedSelDate;
+        var sel;
+
+        return Promise.all([
+           self.collectIpmiSelInformation(data),
+           waterline.workitems.findOne({ id: data.workItemId })
+        ])
+        .spread (function(selInfo, workObj){
+            totalEntries = parseInt(selInfo.Entries);
+            var tmpDate = selInfo["Last Add Time"].split(/\/|:| /);
+            tmpDate =tmpDate[0] +" "+ tmpDate[1] +", "+ tmpDate[2] +" "+ tmpDate[3] +":"+ tmpDate[4]+":"+ tmpDate[5] ;
+            currentSelDate = new Date(tmpDate);
+            //Remove the comment below
+            lastReportedSelEntryID = _.get(workObj, 'config.lastReportedSelEntryID', 0);
+            lastUpdatedSelDate = _.get(workObj, 'config.lastReportedSelEntryID', new Date("01 01, 0100 00:00:00"));
+
+            //resting the counter when the the SEL log has been cleared
+            if(totalEntries < lastReportedSelEntryID){
+                lastReportedSelEntryID = 0;
+            }
+            stringifiedEntries = _.range(lastReportedSelEntryID + 1, totalEntries + 1).join(' ');
+            if( currentSelDate > lastUpdatedSelDate){
+               return self.getSelEntries(data, stringifiedEntries)
+                   .tap(function() {
+                       workObj.config.lastReportedSelEntryID = totalEntries;
+                       workObj.config.lastUpdatedSelDate = currentSelDate;
+                       return workObj.save()
+                   })
+            }
+           else {
+               return
+           }
+        })
+        .then(function (unparsedSel) {
+            return parser.parseSelData(unparsedSel);
+        })
+        .then(function (parsedSel) {
+            //we get the raw sel info for each sel entry in order to decode additional informaiton
+            sel = parsedSel;
+            var rawArr =[];
+            var rawData;
+
+            sel = parsedSel;
+            _.forEach(parsedSel,function (entry) {
+                rawData = self.getRawSelEntry(data, entry["SEL Record ID"])
+                    .then(function(data){
+                        return data
+                    })
+                rawArr.push(rawData)
+            })
+            return Promise.all(rawArr)
+        })
+        .then(function (rawArr) {
+            // we add two properties ("Sensor Type Code" and "Event Type Code") to each sel entry.
+            var decodedSel = {
+                "Event Type Code" : "",
+                "Sensor Type Code" : ""
+            };
+            rawArr.forEach(function (value, index) {
+                var values = [];
+                var rawSelRecord;
+                var mask = 1 << 7 //a mask to clear the 7th bit
+
+                rawArr[index] = value.replace(new RegExp('\r?\n','g'), '');
```

Comment:
If you define this RegExp as it's own variable outside the for loop (e.g. `var regex = new RegExp(...)`), and pass it in like `.replace(regex, '')`, then we'll only compile the regex once rather than for every iteration. Probably an inconsequential performance improvement though.